DataFormats/Detectors/EMCAL/include/DataFormatsEMCAL/Cluster.h

```diff
@@ -28,68 +26,31 @@ namespace emcal
 /// \class Cluster
 /// \brief EMCAL Cluster
 ///
-class Cluster : public o2::dataformats::TimeStamp<Float16_t>, public Point3D<Float16_t>
+class Cluster : public o2::dataformats::TimeStamp<Float16_t>
```

Comment:
Hi @shahor02,

As EMCAL is not a continuous detector the intend we intend to store the time with respect to trigger and not the full time stamp, both for cells and for clusters. This is currently being changed in the digitizer as well! Here we have to cover a range of 1.5 mus with a precision of 1 ns. The cluster time is defined as the time of the leading cell, therefore clusters and cells have the same precision. For cells we currently use 9 bits. This is sufficient as the signal has a delay of ~700 ns which we are going to shift in the raw fitter, so the time range to cover is < 800 ns.

Cheers

Markus

Reply:
Maybe Markus will comment further on this: As far as I understand, EMCAL cells have a time resolution of 10 bits so we need 16 bits.

Reply:
Thanks @mfasDa , in this case, isn't it better to use int16_t for time difference in ns, instead of Float16_t? It will provide the needed precision and will be even compressible (though only a few bits will be gained).